Source/WebCore/platform/audio/cocoa/CAAudioStreamDescription.h

```diff
@@ -80,16 +80,16 @@ class WEBCORE_EXPORT CAAudioStreamDescription final : public AudioStreamDescript
 template<class Encoder>
 void CAAudioStreamDescription::encode(Encoder& encoder) const
 {
-    encoder.encodeFixedLengthData(reinterpret_cast<const uint8_t*>(&m_streamDescription), sizeof(m_streamDescription), 1);
+    encoder.encodeObject(m_streamDescription);
```

Comment:
I think in medium term we want to remove the invocations of `encodeObject`. 
So what I think Alex was saying is that we don't want to have future where the per-member encode / decode logic sometimes calls "encode" and sometimes "encodeObject". For the most part, we want to have regular form.

For purpose of this discussion, encoding / decoding could be divided into three categories:

1. is_trivially_constructible argument coders
2. is_trivially_constructible argument coders that validate
3. per-member argument coders that may or may not validate

For current code that ends up using encodeObject / decodeObject in the 1. category, near term we should instead add a trivially copiable argument coder and then use the regular form (encode <<, decode<>()).

I don't think we have code in category 2. If we did, encodeObject/decodeObject would be a syntactic sugar, less error prone version over encodeSpan/decodeSpan. So in this light, it seems ok to me to add these in this patch since you've written it already. However, if you modify the patch according to the feedback below, then maybe we could try to live without encodeObject/decodeObject with the rationale of not adding yet yet another encode / decode method.


So here we should mark up ideally in this way:

1. AudioStreamBasicDescription is serialised with a trivially copiable argument coder
2. CAAudioStreamDescription is serialised with a member wise argument coder

So we are lacking 1. currently. If we had that, we wouldn't need to call encodeObject.
So we would do:

This would go to somewhere sensible, maybe `WebCoreArgumentEncodersCocoa.h`. The generated serializers don't seem to be able to generate this yet.

```
template<> struct ArgumentCoder<AudioStreamBasicDescription> : SimpleArgumentCoder< AudioStreamBasicDescription> { };
```

For the CAAudioStreamDescription, we could then have a regular by-member encode / decode.
Then in future patch, we would move the CAAudioStreamDescription in some generated serializer .in file.

Reply:
The plan was to take SimpleArgumentCoder, probably bring it into WTF's ArgumentCoder.h as something like `TriviallyCopyableArgumentCoder`, and provide ArgumentCoder specializations for AudioStreamBasicDescription as well as the union type below and I guess also LOGFONT on Windows. That brings everything back into uniformity.